// xendit-node/payment_method/models/GetAllPaymentMethods400Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This file is auto generated by Xendit.
 * Do not edit the class manually.
 * Improvements? Share your ideas at https://github.com/xendit/xendit-node
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAllPaymentMethods400ResponseToJSON = exports.GetAllPaymentMethods400ResponseFromJSONTyped = exports.GetAllPaymentMethods400ResponseFromJSON = exports.instanceOfGetAllPaymentMethods400Response = exports.GetAllPaymentMethods400ResponseErrorCodeEnum = void 0;
const runtime_1 = require("../../runtime");
/**
 * @export
 */
exports.GetAllPaymentMethods400ResponseErrorCodeEnum = {
    AccountAccessBlocked: 'ACCOUNT_ACCESS_BLOCKED',
    AddressValidationFailed: 'ADDRESS_VALIDATION_FAILED',
    AmountMismatched: 'AMOUNT_MISMATCHED',
    ApiValidationError: 'API_VALIDATION_ERROR',
    AuthenticationFailed: 'AUTHENTICATION_FAILED',
    AuthenticationRequired: 'AUTHENTICATION_REQUIRED',
    BrandNotSupported: 'BRAND_NOT_SUPPORTED',
    CardDeclined: 'CARD_DECLINED',
    ChannelCodeNotSupportedError: 'CHANNEL_CODE_NOT_SUPPORTED_ERROR',
    CofCombinationNotAllowedError: 'COF_COMBINATION_NOT_ALLOWED_ERROR',
    CurrencyMismatched: 'CURRENCY_MISMATCHED',
    CustomerNotFoundError: 'CUSTOMER_NOT_FOUND_ERROR',
    CustomerPaymentMethodMismatched: 'CUSTOMER_PAYMENT_METHOD_MISMATCHED',
    DeclinedByIssuer: 'DECLINED_BY_ISSUER',
    DeclinedByProcessor: 'DECLINED_BY_PROCESSOR',
    DuplicatedFixedPaymentInstrument: 'DUPLICATED_FIXED_PAYMENT_INSTRUMENT',
    DuplicateReference: 'DUPLICATE_REFERENCE',
    ExpiredCard: 'EXPIRED_CARD',
    ExpiredOtpError: 'EXPIRED_OTP_ERROR',
    FeatureNotActivated: 'FEATURE_NOT_ACTIVATED',
    InactiveOrUnauthorizedCard: 'INACTIVE_OR_UNAUTHORIZED_CARD',
    InactivePaymentMethod: 'INACTIVE_PAYMENT_METHOD',
    IncorrectAmount: 'INCORRECT_AMOUNT',
    InsufficientBalance: 'INSUFFICIENT_BALANCE',
    InvalidAccountDetails: 'INVALID_ACCOUNT_DETAILS',
    InvalidCvv: 'INVALID_CVV',
    InvalidLinkedAccount: 'INVALID_LINKED_ACCOUNT',
    InvalidOtpError: 'INVALID_OTP_ERROR',
    InvalidPaymentMethod: 'INVALID_PAYMENT_METHOD',
    InvalidToken: 'INVALID_TOKEN',
    LinkedAccountNotFoundError: 'LINKED_ACCOUNT_NOT_FOUND_ERROR',
    MaxAccountLinking: 'MAX_ACCOUNT_LINKING',
    MaxAmountLimitError: 'MAX_AMOUNT_LIMIT_ERROR',
    MaxOtpAttemptsError: 'MAX_OTP_ATTEMPTS_ERROR',
    NotSupportedByChannel: 'NOT_SUPPORTED_BY_CHANNEL',
    OperationNotAllowed: 'OPERATION_NOT_ALLOWED',
    OtpDeliveryError: 'OTP_DELIVERY_ERROR',
    PartnerChannelError: 'PARTNER_CHANNEL_ERROR',
    PaymentMethodNotFoundError: 'PAYMENT_METHOD_NOT_FOUND_ERROR',
    PaymentMethodNotSupported: 'PAYMENT_METHOD_NOT_SUPPORTED',
    PaymentMethodReusabilityInvalid: 'PAYMENT_METHOD_REUSABILITY_INVALID',
    PaymentMethodStatusInvalid: 'PAYMENT_METHOD_STATUS_INVALID',
    PaymentMethodTypeInvalid: 'PAYMENT_METHOD_TYPE_INVALID'
};
/**
 * Check if a given object implements the GetAllPaymentMethods400Response interface.
 */
function instanceOfGetAllPaymentMethods400Response(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfGetAllPaymentMethods400Response = instanceOfGetAllPaymentMethods400Response;
function GetAllPaymentMethods400ResponseFromJSON(json) {
    return GetAllPaymentMethods400ResponseFromJSONTyped(json, false);
}
exports.GetAllPaymentMethods400ResponseFromJSON = GetAllPaymentMethods400ResponseFromJSON;
function GetAllPaymentMethods400ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errorCode': !(0, runtime_1.exists)(json, 'error_code') ? undefined : json['error_code'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
    };
}
exports.GetAllPaymentMethods400ResponseFromJSONTyped = GetAllPaymentMethods400ResponseFromJSONTyped;
function GetAllPaymentMethods400ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error_code': value.errorCode,
        'message': value.message,
    };
}
exports.GetAllPaymentMethods400ResponseToJSON = GetAllPaymentMethods400ResponseToJSON;
